### ISSUE TITLE
MSSQL Update statement with aliases generates incorrect syntax

### ISSUE THREAD

**Issue by kzettlmeier** [hiddentao/squel#244]
When creating an Microsoft SQL Server Update statement using squel.js, we currently get incorrect syntax. Here is an example below:
Squel Statement:
```Javascript
var squel = require('squel');
var mssquel = squel.useFlavour('mssql');
var query = mssquel.update();
query.table("dbo.table", "tbl")
.set("tbl.col1 = 'test'")
.set("tbl.col2 = (0 - tbl.col2)");
.where("tbl.col3 = 'test' AND tbl.col4 = 1");
```
Generated SQL Statement:
```SQL
UPDATE dbo.table tbl 
SET tbl.col1 = 'test', 
tbl.col2 = (0 - tbl.col2) 
WHERE (tbl.col3 = 'test') AND 
(tbl.col4 = 1)
```
This generated SQL Statement is invalid, it should actually be:
```SQL
UPDATE tbl 
SET tbl.col1 = 'test', 
tbl.col2 = (0 - tbl.col2) 
FROM dbo.table tbl
WHERE (tbl.col3 = 'test') AND 
(tbl.col4 = 1)
```

**Comment by hiddentao** [comment]
Looks like the table name aliasing is broken. Is that the main difference between the queries?


On 11 June 2016 at 04:28:40, Kendall Zettlmeier (notifications@github.com) wrote:

When creating an Microsoft SQL Server Update statement using squel.js, we currently get incorrect syntax. Here is an example below:
Squel Statement:

var squel = require('squel');
var mssquel = squel.useFlavour('mssql');
var query = mssquel.update();
query.table("dbo.table", "tbl")
.set("tbl.col1 = 'test'")
.set("tbl.col2 = (0 - tbl.col2)");
.where("tbl.col3 = 'test' AND tbl.col4 = 1");
Generated SQL Statement:

UPDATE dbo.table tbl  
SET tbl.col1 = 'test',  
tbl.col2 = (0 - tbl.col2)  
WHERE (tbl.col3 = 'test') AND  
(tbl.col4 = 1)
This generated SQL Statement is invalid, it should actually be:

UPDATE tbl  
SET tbl.col1 = 'test',  
tbl.col2 = (0 - tbl.col2)  
FROM dbo.table tbl
WHERE (tbl.col3 = 'test') AND  
(tbl.col4 = 1)

**Comment by kzettlmeier** [comment]
Yes, that is the only issue that I can see with the query.  It just needs to add the "FROM tableName alias" under the "SET" statement and use the alias in the "UPDATE" statement and everything should work fine :)

**Comment by hiddentao** [comment]
Ah actually, that's a bigger change than I realised. I'm surprised this hasn't come up before for mssql.
